control-plane/catalog/to-consul/resource.go

```diff
@@ -997,7 +997,7 @@ func (t *serviceIngressResource) Upsert(key string, raw interface{}) error {
 			}
 			hostName = ingress.Status.LoadBalancer.Ingress[0].IP
 		} else {
-			hostName = rule.Host
+			hostName = ingress.Status.LoadBalancer.Ingress[0].Hostname
```

Comment:
Wouldn't it make sense to add a separate if-branch with a config option, `SyncLoadBalancerHostnames` (similar to `SyncLoadBalancerIPs`), to maintain the current behavior? That way, people who need it can enable it, while ensuring we don't break anything.